File: BusinessObjects/Entities/Project.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace forCrowd.Backbone.BusinessObjects.Entities
{
    public class Project : BaseEntity
    {
        public Project()
        {
            ElementSet = new HashSet<Element>();
        }

        public int Id { get; set; }

        public int UserId { get; set; }

        [Required]
        [StringLength(50)]
        public string Name { get => name; set => name = value.Trim(); }

        [Required]
        [StringLength(500)]
        public string Origin { get => origin; set => origin = value.Trim(); }

        [StringLength(5000)]
        public string Description { get => description; set => description = value?.Trim(); }

        public int RatingCount { get; set; }

        public virtual User User { get; set; }
        public virtual ICollection<Element> ElementSet { get; set; }

        string name;
        string origin;
        string description;
    }
}

File: AngularClient/projects/forcrowd-backbone/src/lib/entities/project.ts
import { Element } from "./element";
import { EntityBase } from "./entity-base";
import { User } from "./user";

export class Project extends EntityBase {

  // Server-side
  Id = 0;
  User: User;
  RatingCount = 0;
  ElementSet: Element[];

  get Name(): string {
    return this.fields.name;
  }
  set Name(value: string) {
    this.fields.name = value.trim();
  }

  get Origin(): string {
    return this.fields.origin;
  }
  set Origin(value: string) {
    this.fields.origin = value.trim();
  }

  get Description(): string {
    return this.fields.description;
  }
  set Description(value: string) {
    this.fields.description = value ? value.trim() : null;
  }

  private fields: {
    description: string,
    name: string,
    origin: string,
  } = {
      description: null,
      name: "",
      origin: "",
    };

  initialize(): void {
    if (this.initialized) return;

    super.initialize();

    this.ElementSet.forEach(element => {
      element.initialize();
    });
  }
}
